### PR TITLE
AIO slim and full

### PR DESCRIPTION
**Pull request by JohnFawkes**

### CODE REVIEW

File: common/install.sh [hunk 1]
```diff
@@ -132,33 +184,40 @@ else
...
-OVERLAY='/data/resource-cache/overlays.list'
-if [ -f "$OVERLAY" ]; then
-  ui_print "   Removing $OVERLAY"
-  rm -f "$OVERLAY"
-fi
-
-if [ $API -ge 28 ]; then
-  ui_print " "
-  ui_print "   Enabling Google's Call Screening..."
+if $BOOT;then
```
Comment (thehappydinoa) [inline common/install.sh]: Needs a space? `; then`

File: common/install.sh [hunk 2]
```diff
@@ -113,17 +165,17 @@ if $FUNCTION; then
   else
     ui_print " "
     ui_print "   Enabling overlay features..."
-    sed -i -e 's/ro.boot.vendor.overlay.theme/# ro.boot.vendor.overlay.theme/g' $INSTALLER/common/system.prop
+    sed -i 's/ro.boot.vendor.overlay.theme/# ro.boot.vendor.overlay.theme/g' $INSTALLER/common/system.prop
```
Comment (thehappydinoa) [inline common/install.sh:164]: Do we want to do this with `prop_process`?

File: common/install.sh [hunk 3]
```diff
@@ -67,42 +70,91 @@ ignorewarning() {
...
 fi
 
-if [ "$OOS" ]; then
+#had to break up volume options this way for basename zip for users without working vol keys
+if $SLIM; then 
   ui_print " "
-  ui_print "   Pix3lify has been known to not work and cause issues on devices running OxygenOS!"
-  ignorewarning
+  ui_print "   Enabling slim mode..."
+  rm -f $INSTALLER/common/PhenotypePrefs.xml
```
Comment (thehappydinoa) [inline common/install.sh]: We don't need to delete any content of the `common` folder

File: common/service.sh [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
-MODDIR=${0%/*}
-
 # This script will be executed in late_start service mode
 # More info in the main Magisk thread
+
+if [ $API -ge 28 ] && [ "$FULL" ]; then
+  am force-stop "com.google.android.apps.wellbeing"
+  pm enable "com.google.android.apps.wellbeing/com.google.android.apps.wellbeing.autodnd.ui.AutoDndGesturesSettingsActivity"
+  if [[ $(pm list packages "com.google.android.soundpicker") ]]; then
+  pm grant com.google.android.soundpicker android.permission.READ_EXTERNAL_STORAGE
```
Comment (thehappydinoa) [inline common/service.sh]: Needs a tab

File: common/install.sh [hunk 5]
```diff
@@ -132,33 +184,40 @@ else
...
+      magiskpolicy --live "create system_server sdcardfs file" "allow system_server sdcardfs file { write }"
+      am force-stop "com.google.android.apps.wellbeing"
+    fi
+  fi
+fi
+#add slim & full variables to service.sh
+for i in "SLIM" "FULL"; do
+  sed -i "2i $i=$(eval echo \$$i)" $INSTALLER/common/service.sh
+done
+  cp_ch -n $INSTALLER/common/service.sh $UNITY/service.sh
```
Comment (thehappydinoa) [inline common/install.sh]: Extra tab

**Review by thehappydinoa — DISMISSED** [review]

File: common/install.sh (hunk 2 above)
Comment (JohnFawkes) [reply common/install.sh:164]: prop_process doesn't work that way. The props have to be in a separate file from system.prop.

File: common/install.sh (hunk 3 above)
Comment (JohnFawkes) [reply common/install.sh]: Didn't even see that. @ThatNerdyPikachu had this from the original profiles branch

File: common/install.sh (hunk 1 above)
Comment (JohnFawkes) [reply common/install.sh]: Thanks

File: common/install.sh [hunk 6]
```diff
@@ -132,33 +183,40 @@ else
...
-fi
-
-if [ $API -ge 28 ]; then
-  ui_print " "
-  ui_print "   Enabling Google's Call Screening..."
+if $BOOT; then
   ui_print " "
-  ui_print "   Enabling Google's Flip to Shhh..."
+  ui_print "   Enabling boot animation..."
+  cp -f $INSTALLER/common/bootanimation.zip $UNITY$BFOLDERBZIP
```
Comment (ThatNerdyPikachu) [inline common/install.sh:200]: Is the variable name correct?

File: config.sh [hunk 7]
```diff
@@ -64,6 +64,14 @@ unity_custom() {
   N5X=$(grep -E "ro.product.device=bullhead|ro.product.name=bullhead" $BUILDS)
   N6P=$(grep -E "ro.product.device=angler|ro.product.name=angler" $BUILDS)
   OOS=$(grep -E "ro.product.manufacturer=OnePlus|ro.product.vendor.brand=OnePlus" $BUILDS)
+  MANUFACTURER=$(grep "ro.product.manufacturer" $BUILDS)
+  if [ "$MANUFACTURER" == "HTC" ]; then
+    BFOLDER="/system/customize/resource/"
+    BZIP="hTC_bootup.zip"
```
Comment (ThatNerdyPikachu) [inline config.sh:70]: Is it actually supposed to be hTC?

File: common/install.sh [hunk 8]
```diff
@@ -67,42 +70,90 @@ ignorewarning() {
...
+    fi
+    ui_print " "
+    ui_print " - Animation Options -"
+    ui_print "   Do you want the Pixel boot animation?"
+    ui_print "   Vol Up = Yes, Vol Down = No"
+    if $FUNCTION; then
+      BOOT=true
+    fi
+  else
+    ui_print " Options Specified in Zipname!"
```
Comment (ThatNerdyPikachu) [inline common/install.sh]: ``Options specified in zip name!``

**Review by ThatNerdyPikachu — APPROVED** [review]

File: common/install.sh (hunk 3 above)
Comment (ThatNerdyPikachu) [reply common/install.sh]: Correct, I just did not want to have any unused files.

File: common/install.sh (hunk 6 above)
Comment (JohnFawkes) [reply common/install.sh:200]: Yes Check config for variables.

File: config.sh (hunk 7 above)
Comment (JohnFawkes) [reply config.sh:70]: Unfortunately yes. Htc is weird af

File: common/install.sh (hunk 8 above)
Comment (JohnFawkes) [reply common/install.sh]: Will do

**Comment by ThatNerdyPikachu** [comment]
@lazerl0rd We remove the wallpapers for slim correct? Just making sure.

**Comment by lzlrd** [comment]
Yes @thatnerdypikachu

**Comment by ThatNerdyPikachu** [comment]
Alright it's got my approval, we should wait for @Stallix though.

**Comment by joeyhuab** [comment]
Looks good and clean. Thanks a lot! :) Yeah it's good.

**Review by joeyhuab — APPROVED** [review]

**Comment by joeyhuab** [comment]
Also feel free to add to the readme description on how you guys got added into the project and how you think of the module as a whole before becoming a contributor.

**Comment by ThatNerdyPikachu** [comment]
@Stallix Is everything we removed correct?

**Comment by joeyhuab** [comment]
> @Stallix Is everything we removed correct?

Yep! All good

**Comment by ThatNerdyPikachu** [comment]
Anyone have any objections to a merge?
If not, paging @thehappydinoa to do a changelog and version bump to 2.5 (or anyone else)

**Comment by thehappydinoa** [comment]
@ThatNerdyPikachu I would be happy to.

**Comment by ThatNerdyPikachu** [comment]
:)

**Comment by ThatNerdyPikachu** [comment]
Wait we already have one...

**Comment by lzlrd** [comment]
Don’t u think there’s too mucch customizability here?

**Comment by lzlrd** [comment]
I would say leaving Pix3lify.apk always part of it, an option for slim (Replacing the current no overlays + no accent), and an option for Google theme (sounds, accent, bootanimation) would be good.

**Comment by ThatNerdyPikachu** [comment]
There's not many options, I don't consider it too much.

**Pull request merged**